openslides_backend/action/actions/poll/mixins.py

```diff
@@ -19,16 +20,49 @@ def check_permissions(self, instance: Dict[str, Any]) -> None:
             )
             content_object_id = poll.get("content_object_id", "")
             meeting_id = poll["meeting_id"]
+        check_poll_or_option_perms(
+            self.name, content_object_id, self.datastore, self.user_id, meeting_id
+        )
+
+
+def check_poll_or_option_perms(
+    action_name: str,
+    content_object_id: str,
+    datastore: DatastoreService,
+    user_id: int,
+    meeting_id: int,
+) -> None:
+
+    if content_object_id.startswith("motion" + KEYSEPARATOR):
+        _check_perm(
+            datastore,
+            user_id,
+            Permissions.Motion.CAN_MANAGE_POLLS,
+            meeting_id,
+            action_name,
+        )
+    elif content_object_id.startswith("assignment" + KEYSEPARATOR):
+        _check_perm(
+            datastore,
+            user_id,
+            Permissions.Assignment.CAN_MANAGE,
+            meeting_id,
+            action_name,
+        )
+    else:
+        _check_perm(
+            datastore, user_id, Permissions.Poll.CAN_MANAGE, meeting_id, action_name
+        )
 
-        if content_object_id.startswith("motion" + KEYSEPARATOR):
-            self._check_perm(Permissions.Motion.CAN_MANAGE_POLLS, meeting_id)
-        elif content_object_id.startswith("assignment" + KEYSEPARATOR):
-            self._check_perm(Permissions.Assignment.CAN_MANAGE, meeting_id)
-        else:
-            self._check_perm(Permissions.Poll.CAN_MANAGE, meeting_id)
 
-    def _check_perm(self, perm: Permission, meeting_id: int) -> None:
-        if not has_perm(self.datastore, self.user_id, perm, meeting_id):
-            msg = f"You are not allowed to perform action {self.name}."
-            msg += f" Missing permission: {perm}"
-            raise PermissionDenied(msg)
+def _check_perm(
+    datastore: DatastoreService,
+    user_id: int,
+    perm: Permission,
+    meeting_id: int,
+    action_name: str,
+) -> None:
+    if not has_perm(datastore, user_id, perm, meeting_id):
+        msg = f"You are not allowed to perform action {action_name}."
+        msg += f" Missing permission: {perm}"
+        raise PermissionDenied(msg)
```

Comment:
This does not need to be a separate method, you can just determine the needed permission in the if-else-block above and then move this code up and call `has_perm` with the determined permission.

Reply:
Refactor check_poll_or_option_perm. Remove _check_perm and move perm calculation in the beginning.